src/stcal/resample/resample.py

```diff
@@ -0,0 +1,1506 @@
+                    if self._wcs_pars:
+                        log.info(
+                            "'wcs_pars' will be ignored when 'output' is "
+                            "provided and accum=True"
+                        )
+                except FileNotFoundError:
+                    pass
+
+        elif output is not None:
+            self._output_filename = self.get_model_attr_value(
+                output,
+                "filename"
+            )
+            self._output_model = output
+            self._close_output = False
+
+        return kwargs
+
+    def _create_new_output_model(self):
+        # this probably needs to be an abstract class.
+        # also this is mostly needed for "single" drizzle.
+        output_model = self.new_model(
+            None,
+            copy_meta_from=self._first_model_meta
+        )
+
+        # update meta data and wcs
+        pix_area = self._output_pixel_scale**2
+        self.set_model_meta(
+            output_model,
+            {
+                "wcs": deepcopy(self._output_wcs),
+                "pixelarea_steradians": pix_area,
+                "pixelarea_arcsecsq": pix_area * np.rad2deg(3600)**2,
+            }
+        )
+
+        return output_model
+
+    def build_output_model_name(self):
+        fnames = {f for f in self._input_filename_list if f is not None}
+
+        if not fnames:
+            return "resampled_data_{resample_suffix}{resample_file_ext}"
+
+        # TODO: maybe remove ending suffix for single file names?
+        prefix = os.path.commonprefix(
+            [PurePath(f).stem.strip('_- ') for f in fnames]
+        )
+
+        return prefix + "{resample_suffix}{resample_file_ext}"
+
+    def create_output_model(self, resample_results):
+        # this probably needs to be an abstract class (different telescopes
+        # may want to save different arrays and ignore others).
+
+        if not self._output_model and self._output_filename:
+            if self._accum and Path(self._output_filename).is_file():
+                self._output_model = self.open_model(self._output_filename)
+            else:
+                self._output_model = self._create_new_output_model()
+                self._close_output = not self.in_memory
+
+        if self._output_filename is None:
+            self._output_filename = self.build_output_model_name()
+
+        self.set_model_array(self._output_model, "data", resample_results.out_img)
+
+        self.update_exposure_times()
+        if self._enable_err:
+            self._finish_variance_processing()
+
+        self.set_model_meta(
+            self._output_model,
+            {
+                "weight_type": self.weight_type,
+                "pointings": len(self._input_models.group_names),
+            }
+        )
+        # TODO: also store the number of images added in total: ncoadds?
+
+        self.final_post_processing()
+
+        self._output_model.write(self._output_filename, overwrite=True)
```

Comment:
This won't work for roman since `write` does not exist for `roman_datamodels.DataModel` and `overwrite` is not a valid keyword argument.

I think it's best if we leave all file-IO up to the pipeline and not include it here in stcal.

Reply:
that was an omission. I'll fix this.